code/modules/roguetown/roguejobs/blacksmith/smelter.dm

```diff
@@ -77,9 +80,11 @@
 				if(cooking == 20)
 					for(var/obj/item/I in ore)
 						if(I.smeltresult)
+							while(I.smelt_bar_num)
+								I.smelt_bar_num--
+								var/obj/item/R = new I.smeltresult(src, ore[I])
+								ore += R
 							ore -= I
-							var/obj/item/R = new I.smeltresult(src)
-							ore += R
```

Comment:
again, im not against smelt bar num existing, just not having weapons smelt for the same amount of resources they take to smith!